RimeWithWeasel/RimeWithWeasel.cpp

```diff
@@ -28,6 +28,37 @@ typedef enum { COLOR_ABGR = 0, COLOR_ARGB, COLOR_RGBA } ColorFormat;
 #endif
 using namespace weasel;
 
+template <typename CharT>
+inline std::basic_string<CharT> escape_string(
+    const std::basic_string<CharT>& str) {
+  std::basic_string<CharT> res{};
+  CharT backslash;
+  CharT newline;
+  if constexpr (std::is_same_v<CharT, wchar_t>) {

```

Comment:
Maybe it's simpler to define another type and replace the type check.
Try to make the following code work; I'm not sure.

``` c++
template <CharT>
struct Escape {
  static constexpr CharT backslash;
  static constexpr CharT newline;
};

template <>
struct Escape<char> {
  static constexpr char backslash = '\\';
  static constexpr char newline = '\n';
};

template <>
struct Escape<wchar_t> {
  static constexpr wchar_t backslash = L'\\';
  static constexpr wchar_t newline = L'\n';
};

// use it in a template function
CharT ch = Escape<CharT>::backslash;